### PR TITLE
subsys: settings: Fix Kconfig dependencies

### PR DESCRIPTION
**Pull request by jhedberg**
If CONFIG_FLASH_PAGE_LAYOUT is not enabled, building the settings code
fails at the linking stage with the following error when FCB is the
backend.

```
libzephyr.a(settings_init.c.obj): In function `settings_init_fcb':
subsys/settings/src/settings_init.c:62: undefined reference to `flash_area_get_sectors'
collect2: error: ld returned 1 exit status
```

Also, the File System backend will currently only work with NFFS due
to fs_rename() missing from other File Systems (FAT in particular).

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

### CODE REVIEW

**Comment by jhedberg** [comment]
Another question: NFFS does `select FLASH_PAGE_LAYOUT` instead of `depends on FLASH_PAGE_LAYOUT`. Should I also use select for SETTINGS? What's the reasoning for choosing one over the other?

**Comment by jhedberg** [comment]
@andrzej-kaczmarek could you comment on the above questions? `git blame` seems to point at you at least regarding the NFFS Kconfig entry :)

**Comment by andrzej-kaczmarek** [comment]
@jhedberg this was according to review comment [1] but tbh I do not remember the exact reason for this. but since comment was made by @carlescufi perhaps he knows the answer :)

[1] https://github.com/zephyrproject-rtos/zephyr/pull/1288#pullrequestreview-63636842

**Comment by jhedberg** [comment]
Note regarding the CI failure: it's a false-positive due to a "too long" commit message line of a verbatim copy-pasted line from the build error.

File: subsys/settings/Kconfig [hunk 1]
```diff
@@ -9,6 +9,7 @@ menuconfig SETTINGS
 	default n
 	prompt "Enable settings subsystem with non-volatile storage"
 	depends on FILE_SYSTEM || FCB
+	depends on FLASH_PAGE_LAYOUT
```
Comment (nvlsianpu) [inline subsys/settings/Kconfig]: `depends on FILE_SYSTEM || (FCB && FLASH_PAGE_LAYOUT)` 
+
under `config SETTINGS_FCB` change dependency to `(FCB && FLASH_PAGE_LAYOUT)`

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

**Comment by nvlsianpu** [comment]
It is possible to use any FS supported by  VFS along with the settings, but so far only NFFS implements API 
for file rename. Function on which compilation fails is dedicated to initialize while FCB is the settings back-end.

File: subsys/settings/Kconfig (hunk 1 above)
Comment (nvlsianpu) [reply subsys/settings/Kconfig]: Finally i recommend change to:
# Only NFFS is supported as FS.
#The reason ins that FatFs doesn't implements fs_rename() API 
depends on (FILE_SYSTEM && FILE_SYSTEM_NFFS) || (FCB && FLASH_PAGE_LAYOUT)

**Comment by carlescufi** [comment]
>Another question: NFFS does select FLASH_PAGE_LAYOUT instead of depends on FLASH_PAGE_LAYOUT. Should I also use select for SETTINGS? What's the reasoning for choosing one over the other?

The only reason was to ensure that NFFS was the top-level Kconfig option to enable, and that would select automatically the flash page layout, which at least at the time was optional.

File: subsys/settings/Kconfig (hunk 1 above)
Comment (jhedberg) [reply subsys/settings/Kconfig]: fixed

File: subsys/settings/Kconfig [hunk 2]
```diff
@@ -8,7 +8,9 @@ menuconfig SETTINGS
 	bool
 	default n
 	prompt "Enable settings subsystem with non-volatile storage"
-	depends on FILE_SYSTEM || FCB
+	# Only NFFS is currently supported as FS.
+	# The reason ins that FatFs doesn't implement the fs_rename() API
```
Comment (nvlsianpu) [inline subsys/settings/Kconfig]: ins -> is

Comment (jhedberg) [reply subsys/settings/Kconfig]: That was copy-paste from your suggestion ;) Fixing in a moment.

Comment (jhedberg) [reply subsys/settings/Kconfig]: Fixed now

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**